numpy/distutils/mingw32ccompiler.py

```diff
@@ -158,11 +159,11 @@ def __init__ (self,
                 # build needs '-mincoming-stack-boundary=2' due to ABI
                 # incompatibility to Win32 ABI
                 self.set_executables(
-                    compiler='gcc -O2 -march=core2 -mtune=generic'
-                             ' -mfpmath=sse -msse2'
+                    compiler='gcc -O2 -march=pentium4 -mtune=generic'
```

Comment:
```-march=pentium4``` garantuees that no SSE3 instructions or above will be created. This is not the case with ```core2```. 
Please keep in mind, that the most performance critical part of numpy is handled be the OpenBLAS library. This library chooses at runtime the best availabe instruction set according to the CPU type and supports multithreaded execution of BLAS functions as well.